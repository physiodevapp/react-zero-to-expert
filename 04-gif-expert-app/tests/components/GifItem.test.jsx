const { render, screen } = require("@testing-library/react");
const { default: GifItem } = require("../../src/components/GifItem");

describe("testing <GiftItem/>", () => {
  const title = "Goku";
  const url = "https://www.goku.org/";
  
  test("should match component snapshot", () => {
    const {container} = render(<GifItem title={title} url={url}/>);

    expect(container).toMatchSnapshot()
  });

  test('should show url image and alt', () => { 

    render(<GifItem title={title} url={url}/>);

    const {src, alt} = screen.getByRole('img')

    expect(src).toBe(url);
    expect(alt).toBe(title)
   });

   test('should show title ', () => { 
    
    render(<GifItem title={title} url={url}/>);

    expect(screen.getByText(title)).toBeTruthy()

    })
});