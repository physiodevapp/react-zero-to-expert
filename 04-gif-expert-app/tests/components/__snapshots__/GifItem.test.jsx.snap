// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <GiftItem/> should match component snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://www.goku.org/"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
